### PR TITLE
Add Planck keymap

### PR DESCRIPTION
**Pull request by mwpeterson**

### CODE REVIEW

File: keyboards/planck/keymaps/mwpeterson/config.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/planck/keymaps/mwpeterson/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch?

File: keyboards/planck/keymaps/mwpeterson/config.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#define CONFIG_USER_H
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/mwpeterson/config.h]: This line is not actually needed, and should be removed. 

Specifically, this should be included in the keyboard's `config.h` file, and is redundant.

File: keyboards/planck/keymaps/mwpeterson/config.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/planck/keymaps/mwpeterson/config.h]: This is the new default, and isn't actually needed.

File: keyboards/planck/keymaps/mwpeterson/config.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+#define PERMISSIVE_HOLD
+#define IGNORE_MOD_TAP_INTERRUPT
+
+#define NO_DEBUG
```
Comment (drashna) [inline keyboards/planck/keymaps/mwpeterson/config.h]: This is automatically defined if `CONSOLE_ENABLE` is set to "no".

File: keyboards/planck/keymaps/mwpeterson/keymap.c [hunk 5]
```diff
@@ -0,0 +1,305 @@
+// from <https://github.com/noahfrederick/dots/tree/master/planck
+
+#include "planck.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/mwpeterson/keymap.c]: Could you replace this include with `#include QMK_KEYBOARD_H` instead? 

And could you remove the other includes (except for "version.h")?

File: keyboards/planck/keymaps/mwpeterson/keymap.c [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+  unregister_code(PV_RG);
+}
+
+uint32_t layer_state_set_user(uint32_t state) {
+  return update_tri_layer_state(state, LOWER_LAYER, RAISE_LAYER, ADJUST_LAYER);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
```
Comment (drashna) [inline keyboards/planck/keymaps/mwpeterson/keymap.c]: This isn't actually needed, if you only have the one base layer.

**Comment by mwpeterson** [comment]
Requested changes have been made.
Thanks for the feedback!

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**